### DISCUSSION TITLE
Juno for Pulsar

### DISCUSSION THREAD

**Discussion by stefaniecg** [pulsar-edit/.github#289] [Community Package Help]
I have been unsuccessful on installing Juno [1] for Pulsar. 
Does anyone know if this is somehow possible? 
i really do not want to use Visual Studio Code. 

Thank you for the help you can provide. 

Ref. 
[1] https://junolab.org

**Comment by savetheclocktower** [comment]
~~I'm not sure what Juno is. Is this a specific package you were trying to install?~~

EDIT: I thought you might've meant that. [julia-client](https://web.pulsar-edit.dev/packages/julia-client) fails to build, but I think it might be as simple as updating a dependency. Let me take a look.

**Reply by stefaniecg** [reply]
sorry for not putting a raference to what i was talking about, please see the post, i've updated it

**Comment by savetheclocktower** [comment]
I don't have a great answer for you here, but I want to make sure it's actionable. I played around with `uber-juno` and the things it installs and I managed to get past all of the errors; it's possible you'd be able to do the same.

First, on `julia-client`: the error I get during installation can be fixed by updating the `node-pty-prebuilt-multiarch` package, but the whole point is that it won't install in the first place, so you'd have to

* Check it out from GitHub into a directory with `git clone git@github.com:JunoLab/atom-julia-client.git`
* `cd atom-julia-client`
* Update the `package.json` to change `node-pty-prebuilt-multiarch` from `0.9.0` to `^0.10.0`
* `npm install` (should complete successfully; if `npm` isn't found, try `ppm install`)
* `ppm link .`
* `npx electron-rebuild -v 12.2.3` (it might have to install `electron-rebuild`; that's fine — or, if `npx` isn't found, try `ppm rebuild` instead)

These instructions worked for me with `npm`/`npx`, but this presumes you have Node installed locally. The `ppm` versions of those commands may or may not work; I haven't tried. If this part fails, let me know.

The other packages installed by `uber-juno` install successfully, but some of them throw errors on launch. Luckily, there are updated versions of these dependencies on GitHub, so you should uninstall `uber-juno` and `ink` (if they're already installed) and instead do

```
ppm install git@github.com:JunoLab/atom-ink.git
ppm install git@github.com:JunoLab/uber-juno.git
```

and those should be sorted.

---

When I did these things, I wound up with what _seems_ like a functional Julia environment, but I'm no expert. The REPL seems to work, as does autocompletion. I don't love packages that install a toolbar (as though you're only going to be using Pulsar for Julia and not for other things as well!) but I bet that can be disabled without much ill effect.

@mjrodgers has been working on a new Julia language grammar (for updated syntax highlighting) and I'm curious whether he's got suggestions about a good Julia workflow inside of Pulsar. @stefaniecg, tell us what features you're looking for out of a Julia IDE and we may be able to figure out alternative packages that would do the same things.

**Comment by mjrodgers** [comment]
I need to update some things with my tree sitter package, they made a big update to the parser so I want to incorporate those changes.

But as far as getting Juno to work, I had to make my own version of `atom-julia-client` that basically made the changes mentioned above.
[link to my package](https://github.com/mjrodgers/atom-julia-client)

I'm able to use a lot of the features including the REPL and the debugger even. But I would avoid using the automatic code formatting, it uses a very very old version of `JuliaFormatter.jl` that will not properly respect your configuration file. It's impossible to update this dependency unfortunately without major changes to the Juno codebase.

**Reply by savetheclocktower** [reply]
Aha! @stefaniecg, you can give this this version a shot via…

```
ppm install mjrodgers/atom-julia-client
```

…instead of following the instructions above from my bulleted list.

**Reply by mjrodgers** [reply]
Yes! Although the advice about installing `uber-juno` and `ink` from the github links still stands.
